src/test/article.spec.js

```diff
@@ -1197,5 +1199,255 @@ describe('GET /api/v1/articles/popular', () => {
+      .set('token', token)
+      .send({ content: body });
+    expect(res.status).to.equal(201);
+    expect(res.body.message).to.be.equal('Commented successfully');
+    expect(res.body.data.content).to.be.equal(
+      createdComment[0].dataValues.content
+    );
+  });
+
+  it('It should fail if article does not exist', async () => {
+    const comenter = await createUser();
+    const author = await createUser();
+    const createdArticle = await createArticle(
+      await generateArticle({ authorId: author.id })
+    );
+    const token = helper.jwtSigner({ id: comenter.id });
+    const res = await server()
+      .post(`/api/v1/comments/34567`)
+      .set('token', token)
+      .send({ content: createdArticle.body });
+    expect(res.status).to.equal(404);
+    expect(res.body.message).to.be.equal('The requested article was not found');
+  });
+
+  it('should not create a comment if the content body is not provided', async () => {
+    const comenter = await createUser();
+    const author = await createUser();
+    const createdArticle = await createArticle(
+      await generateArticle({ authorId: author.id })
+    );
+    const token = helper.jwtSigner({ id: comenter.id });
+    const res = await server()
+      .post(`/api/v1/comments/${createdArticle.id}`)
+      .set('token', token)
+      .send({ content: '' });
+    expect(res.status).to.equal(400);
+    expect(res.body.message).to.be.equal(
+      'Please provide body for your comment with minimum of 3 characters'
+    );
+  });
+  it('should not create a comment if unauthenticated', async () => {
+    const comenter = await createUser();
+    const author = await createUser();
+    const createdArticle = await createArticle(
+      await generateArticle({ authorId: author.id })
+    );
+    const fakeToken = '1234rtgfdgh';
+    const res = await server()
+      .post(`/api/v1/comments/${createdArticle.id}`)
+      .set('token', fakeToken)
+      .send({ content: createdArticle.body });
+    expect(res.status).to.equal(401);
+    expect(res.body.message).to.be.equal('Token is not valid');
+  });
+});
+describe('GET api/v1/:articleId/comments', () => {
+  beforeEach(async () => {
+    await db.Article.destroy({ cascade: true, truncate: true });
+    await db.User.destroy({ cascade: true, truncate: true });
+  });
+  it('should list all comments for an article', async () => {
+    const comenter = await createUser();
+    const author = await createUser();
+    const createdArticle = await createArticle(
+      await generateArticle({ authorId: author.id })
+    );
+    const token = helper.jwtSigner({ id: comenter.id });
+    const res = await server()
+      .get(`/api/v1/comments/${createdArticle.id}`)
+      .set('token', token);
+    expect(res.status).to.equal(200);
+    expect(res.body.message).to.be.equal('Operation Successful');
+    expect((res.body.data.rows[0] = 1));
+  });
+
+  it('should not list  when comment token is invalid', async () => {
```

Comment:
This behaviour is not right. Users should be able to see comments whether or not they're logged in.